File: be/src/main/java/nl/zzpmatcher/userLogon/controller/UserController.java
package nl.zzpmatcher.userLogon.controller;

import nl.zzpmatcher.userLogon.business.User;
import nl.zzpmatcher.userLogon.business.UserManagement;
import nl.zzpmatcher.userLogon.business.UserRegisterCommand;
import nl.zzpmatcher.userLogon.business.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.hateoas.Resource;
import org.springframework.hateoas.Resources;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpServletResponse;

@RestController
public class UserController {

    private final UserManagement userManagement;

    private final UserRepository userRepository;
    private final UserResourceAssembler userResourceAssembler;

    @Autowired
    public UserController(UserRepository userRepository, UserResourceAssembler userResourceAssembler) {
        this.userRepository = userRepository;
        this.userResourceAssembler = userResourceAssembler;

        this.userManagement = new UserManagement(userRepository);
    }

    @ExceptionHandler(UserManagement.DuplicateEmailAddressException.class)
    @ResponseStatus(value = HttpStatus.CONFLICT, reason = "Double!!")
    public void handleDataFormatException(UserManagement.DuplicateEmailAddressException ex, HttpServletResponse response) {

    }

    @CrossOrigin(origins = "http://local.zzpmatcher.nl:4200")
    @PostMapping("/public/createUser")
    public Resource postCreateUser(@RequestBody UserRegisterCommand userRegisterCommand) {

        User createdUser = userManagement.createUser(userRegisterCommand);

        return userResourceAssembler.toEmailAddressOnlyProjection(createdUser);
    }

    @GetMapping("/admin/user")
    public HttpEntity<Resources<User>> getUser() {

        final Iterable<User> users = userRepository.findAll();

        Resources<User> userResource = userResourceAssembler.toResources(users);
        return ResponseEntity.ok(userResource);
    }
}

File: be/src/main/java/nl/zzpmatcher/userlogon/controller/UserController.kt
package nl.zzpmatcher.userlogon.controller

import nl.zzpmatcher.userlogon.business.User
import nl.zzpmatcher.userlogon.business.UserManagement
import nl.zzpmatcher.userlogon.business.UserRegisterCommand
import nl.zzpmatcher.userlogon.business.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.hateoas.Resource
import org.springframework.hateoas.Resources
import org.springframework.http.HttpEntity
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

import javax.servlet.http.HttpServletResponse

@RestController
class UserController @Autowired
constructor(private val userRepository: UserRepository, private val userResourceAssembler: UserResourceAssembler) {

    private val userManagement: UserManagement

    init {

        this.userManagement = UserManagement(userRepository)
    }

    @ExceptionHandler(UserManagement.DuplicateEmailAddressException::class)
    @ResponseStatus(value = HttpStatus.CONFLICT, reason = "Double!!")
    fun handleDataFormatException(ex: UserManagement.DuplicateEmailAddressException, response: HttpServletResponse) {

    }

    @PostMapping("/public/createUser")
    fun postCreateUser(@RequestBody userRegisterCommand: UserRegisterCommand): Resource<*> {

        val createdUser = userManagement.createUser(userRegisterCommand)

        return userResourceAssembler.toEmailAddressOnlyProjection(createdUser)
    }

    val user: HttpEntity<Resources<User>>
        @GetMapping("/admin/user")
        get() {

            val users = userRepository.findAll()

            val userResource = userResourceAssembler.toResources(users)
            return ResponseEntity.ok(userResource)
        }

    // TODO: voor debug acties, daarna verwijderen
    val allProfiles: Iterable<User>
        @GetMapping("/public/all")
        get() = userRepository.findAll()
}
